torch/_functorch/partitioners.py

```diff
@@ -1011,69 +1011,35 @@ def default_partition(
     Returns:
         Returns the generated forward and backward Fx graph modules.
     """
-    if has_recomputable_ops(joint_module):
-        return min_cut_rematerialization_partition(
-            joint_module,
-            _joint_inputs,
-            num_fwd_outputs=num_fwd_outputs,
-            static_lifetime_input_indices=static_lifetime_input_indices,
-        )
     primal_inputs = list(filter(_is_primal, joint_module.graph.nodes))
     fwd_seed_offset_inputs = list(filter(_is_fwd_seed_offset, joint_module.graph.nodes))
     inputs = primal_inputs + fwd_seed_offset_inputs
     fwd_outputs, bwd_outputs, fwd_outputs_descs, bwd_outputs_descs = (
         _extract_fwd_bwd_outputs(joint_module, num_fwd_outputs=num_fwd_outputs)
     )
-    forward_only_graph = _extract_graph_with_inputs_outputs(
-        joint_module.graph, inputs, fwd_outputs, fwd_outputs_descs, "forward"
-    )
+    # It is NOT sufficient to extract some forward-only graph because can DCE some nodes
+    # that may be important to save. TODO: link to Brian's PR.
+    not_backward_nodes = [
+        x for x in joint_module.graph.nodes if not _has_tag_is_backward(x)
+    ]
```

Comment:
Animesh also mentions offline that this can be achieved by inserting a dummy node between the end of forward and start of backward, but since we have `set_partitioner_tag_is_backward` wrapping the `.grad` call in `create_joint` I'm going to reuse those tags for now.